### ISSUE TITLE
Sync upgrade failing 1.4.0 

### ISSUE THREAD

**Issue by fantawttw** [xBrowserSync/App#41]
### Steps to reproduce
1. Upgrade to 1.4.0 with private server version 1.1.4 running latest code as of 12/10/2018 (https://xbrowsersync.burrows.xyz)

### Expected behaviour
Sync data should be upgraded

### Actual behaviour
Receive error
"Something went wrong
If this problem recurs, submit an issue at github.com/xBrowserSync/App."
Screenshot attached
![error](https://user-images.githubusercontent.com/340697/46859351-6df91980-ce05-11e8-95cb-6316293a5b61.jpg)

### Client app information
**App Platform:** Version 69.0.3497.100 (Official Build) (64-bit)
Andoird

**App Platform Version:** 1.4.0 from Chrome store

#### Other info
```
Debug log from within Chrome app
INFO: Sync data: {"type":4,"command":1,"deferred":{"promise":{"$$state":{"status":2,"value":{},"processScheduled":false,"pur":true}}}}

Server data returned
/info {"maxSyncSize":1048576,"message":"Donate bitcoin: 1LQJnXoyftB6RDJjnv2AoNbNBZVoRbdLhw","status":1,"version":"1.1.4"}
/bookmarks/SYNCID {"bookmarks":"****REDACTED but encoded bookmark data****","lastUpdated":"2018-10-11T08:17:25.467Z"}
```

If you need anymore info let me know.

Cheers

**Comment by nero120** [comment]
@fantawttw could you paste any error messages that occur when you attempt to upgrade here please?

You can find them by browsing to `chrome://extensions/`, then click on `background.html` within the xBrowserSync section. Clear the console when the dev tools window opens, then try running the upgrade within xBrowserSync. You should then see error messages when the red "Something went wrong" bar appears in the app. Paste them in here and I'll try to see what's going on.

**Comment by fantawttw** [comment]
Hi @nero120,
All that appears in the background.html is the following.
Sync data: {"type":4,"command":1,"deferred":{"promise":{"$$state":{"status":2,"value":{},"processScheduled":false,"pur":true}}}}

![image](https://user-images.githubusercontent.com/340697/46937976-76965d80-d05a-11e8-85d7-9044f1a5914b.png)

**Comment by cjnaz** [comment]
Similar to issue #44

**Comment by vsatmydynipnet** [comment]
don't want to open a new issue, because its most likely the same:

runnin 1.1.4 server side and had a workind 1.3.1 (F-Droid) running on my tablet. switched o 1.4.0 from google play via yalp. 

trying to change the service to my servers says "no valid xbrowsersync service"

going back to 1.3.1 meanwhile

**Comment by nero120** [comment]
Sorry guys, been at the hospital for the last week so not had a chance to look into this yet. Hopefully will get a chance next week.

**Comment by nero120** [comment]
@fantawttw I've done some testing upgrading syncs from v1.3.1 to v1.4.0 using your https://xbrowsersync.burrows.xyz service, however each time the upgrade finishes successfully. I've tested with a number of different bookmark data scenarios, but I think the issue is related specifically to your bookmarks data configuration (revealing a bug in the upgrade process).

Would you be willing to [email me](mailto:admin@xbrowsersync.org) a backup of your v1.3.1 sync data so I can use it to debug the upgrade process?

**Comment by nero120** [comment]
@fantawttw I've fixed a bug in the sync upgrade process and produced a new [beta release](https://github.com/xBrowserSync/App/releases/tag/v1.4.1_BETA_2) that includes the fix. Could you download and install the .crx to test whether this fix resolves your issue?

**Comment by nero120** [comment]
I believe this issue has been fixed in commit b17d5d7deffff0d5e2c898d3320ff52b8d28355f and will be released in v1.4.1.

@fantawttw if you can test the fix using the [beta release](https://github.com/xBrowserSync/App/releases/tag/v1.4.1_BETA_2) then please respond here with the results, otherwise you can wait for the full release of v1.4.1. Closing now but feel free to reply here if necessary.

**Comment by nero120** [comment]
Great stuff, thanks for letting me know! 😄
